src/sage/combinat/partition.py

```diff
@@ -2485,6 +2485,122 @@ def conjugate(self):
         par = Partitions_n(sum(self))
         return par.element_class(par, conjugate(self))
 
+    def glaisher_franklin(self, s):
+        r"""
+        Apply the Glaisher-Franklin bijection to ``self``.
+
+        INPUT:
+
+        - ``s`` -- positive integer
+
+        OUTPUT:
+
+        The Franklin-Glaisher bijection, with parameter `s`, returns
+        a partition whose set of parts that are repeated at least `s`
+        times equals the set of parts divisible by `s` in ``self``,
+        after dividing each part by `s`.
+
+        EXAMPLES::
+
+            sage: Partition([4, 3, 2, 2, 1]).glaisher_franklin(2)
+            [3, 2, 2, 1, 1, 1, 1, 1]
+
+        TESTS:
+
+        The map preserves the size::
+
+            sage: all(mu.glaisher_franklin(s).size() == n
+            ....:     for n in range(20) for mu in Partitions(n)
+            ....:     for s in range(1, 5))
+            True
+
+        The map is bijective::
+
+            sage: l = [[mu.glaisher_franklin(s)
+            ....:      for n in range(20) for mu in Partitions(n)]
+            ....:     for s in range(1, 5)]
+            sage: all(len(set(ls)) == len(ls) for ls in l)
+            True
+
+        The map transports the statistics::
+
+            sage: d = lambda la, s: set(p / s for p in la if p % s == 0)
+            sage: r = lambda la, s: set(p for p in la if list(la).count(p) >= s)
+            sage: all(d(mu, s) == r(mu.glaisher_franklin(s), s)
+            ....:     for n in range(20) for mu in Partitions(n)
+            ....:     for s in range(1, 5))
+            True
+
+        For `s=2`, the map is known to findstat::
+
+            sage: findmap(Partitions, lambda mu: mu.glaisher_franklin(2))       # optional - internet
+            0: Mp00312 (quality [100])
+        """
+        s = ZZ(s)
+        if s.is_one():
+            return self
+        mu = []
+        for p, m in enumerate(self.to_exp(), 1):
+            if not p % s:
+                mu.extend([p // s]*(m*s))
+            else:
+                mu.extend(p1 for i, v in enumerate(m.digits(s))
+                          if (p1 := p * s**i)
+                          for _ in range(v))
```

Comment:
```suggestion
                mu.extend([p // s] * (m*s))
            else:
                mu.extend([p1]*v for i, v in enumerate(m.digits(s))
                          if (p1 := p * s**i))
```
The first is PEP8, the second is because it is faster:
```
sage: def test1():
....:     L = [1,2,3]
....:     L.extend([1] * 100)
....:
sage: def test2():
....:     L = [1,2,3]
....:     L.extend(1 for _ in range(100))
....:
sage: %timeit test1()
1.11 µs ± 71.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
sage: %timeit test2()
9.21 µs ± 652 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
Wow, I missed that one!  Thank you!